### PR TITLE
Add setcontext to mo file

### LINKED ISSUE #3441: setcontext is not working correctly for mounit
Calling setcontext on mounit does currently nothing as it inherits code from base class:

``` python
    def setcontext(self, context):
        """Set the message context"""
        pass
```

I'd expect it to properly update context as it does for other storages.

### PR DESCRIPTION
**Pull request by nijel**
To make the API consistent.

Fixes #3441

### CODE REVIEW

File: translate/storage/test_mo.py [hunk 1]
```diff
@@ -124,6 +128,14 @@ def test_language(self):
         store.updateheader(add=True, Language="zu")
         assert store.gettargetlanguage() == "zu"
 
+    def test_context(self):
+        store = self.StoreClass()
+        unit = mo.mounit('source')
```
Comment (unho) [inline translate/storage/test_mo.py]: Nit: Use `store.UnitClass` instead of `mo.mounit`.

**Comment by unho** [comment]
Other than raised nit and styling issue picked up by Travis, lgtm.

File: translate/storage/test_mo.py (hunk 1 above)
Comment (nijel) [reply translate/storage/test_mo.py]: `AttributeError: 'TestMOFile' object has no attribute 'UnitClass'`

Should I add it or keep using mo.mounit?

Comment (nijel) [reply translate/storage/test_mo.py]: In the end I've chosen to use `self.StoreClass.UnitClass`...

**Pull request merged**